src/components/dialog/index.js

```diff
@@ -71,10 +72,32 @@ class Dialog extends BaseComponent {
     };
   }
 
+  onSwipe(gestureName) {
+    const {SWIPE_UP, SWIPE_DOWN} = swipeDirections;
+    switch (gestureName) {
+      case SWIPE_UP:
+        if (this.props.top) {
```

Comment:
I wouldn't use the top modifier to decide how and if to swipe, the modifiers are "nice to have" props and not mandatory. 
User can align his dialog using simple style. 
I would expose a dedicated prop for that behaviour (swipeToDismissDirection or something like that)